kdd-2024/ibis-tutorial-KDD2024.md

```diff
@@ -0,0 +1,273 @@
+# Descriptive Title 
+
+We need a title here, in the past we had this options
+
+- Scipy and PyCon 2024: Intro to Ibis: blazing fast analytics with DuckDB, Polars, Snowflake, and more,
+from the comfort of your Python repl.
+- PyData NYC 2023: Ibis - Expressive analytics in Python at any scale
+
+I think we needs something more DS/ML/AI kind of thing. Here are some options I 
+brainstormed with gpt and modified, but Im not convinced by anyone: 
+
+- "Ibis: The Data Scientist's Powerhouse for Blazing-Fast Analytics and Machine Learning"
+- "Mastering Data Analytics and ML with Ibis: Speed, Simplicity, and Scale"
+- "Elevate Your Data Science Game with Ibis: Fast, Flexible, and Future-Proof"
+- "Ibis: Your Gateway to Expressive Analytics and Streamlined ML"
+
+# Abstract (300 words)
+
+I tried to adapt what we had for SciPy and PyCon, and include something on Streaming 
+and ML. Keep in mind that streaming and ML will be a small part of this tutorial, per 
+Ian's suggestion 20%, please make suggestions with this in mind. 
+
+**(I believe we are at 270 words now)**
+
+Tabular data analysis, a staple in data science, often encounters scalability 
+challenges with pandas. In response, modern analytical databases (like DuckDB) 
+offer significant performance boosts but typically demand SQL expertise. Many of 
+these systems only provide a SQL interface though; something far different from 
+pandas’ dataframe interface, requiring a rewrite of your analysis code.
+
+This is where Ibis comes in. Ibis is a pure-Python open-source library, originally 
+created by Wes McKinney creator of pandas, that provides a dataframe interface to
+many popular databases and analytics tools (DuckDB, Polars, Snowflake, Spark, etc). This 
+lets users analyze data using the same consistent API, regardless of which backend 
+they’re using, and without ever having to learn SQL. This tutorial introduces 
+Ibis, emphasizing its role in simplifying data manipulation across diverse platforms.
+
+Additionally, attendees will explore Ibis' advancements in data streaming technology. The 
+Ibis-flink backend, inspired by Apache Flink SQL, combines stream and batch into a single 
+framework, addressing common barriers faced by data scientists venturing into 
+streaming analytics. By leveraging Ibis unified Python dataframe API, we facilitate 
+seamless transitions between batch and streaming workflows, reducing operational
+complexities and enabling real-time machine learning applications.
+
+Last but not least, this tutorial will also introduce Ibis ML, an evolving library 
+for building machine learning pipelines using Ibis. Where these pipelines can then
```

Comment:
Ibis ML -> IbisML

Perhaps in place of the second sentence, something like:

"IbisML brings the portability of Ibis to ML pipelines -- preprocess your data on any of the 20+ backends Ibis supports, then efficiently handoff to popular training libraries like scikit-learn, XGBoost, or PyTorch."